local/src/core/http.cc

```diff
@@ -1227,8 +1231,103 @@ Session::set_fd(int fd)
   return errata;
 }
 
+InterfaceNameToEndpoint::InterfaceNameToEndpoint(TextView expected_interface, int expected_family)
+  : _expected_interface{expected_interface}
+  , _expected_family{expected_family}
+{
+}
+
+InterfaceNameToEndpoint::~InterfaceNameToEndpoint()
+{
+  if (_ifaddr_list_head != nullptr) {
+    freeifaddrs(_ifaddr_list_head);
+    _ifaddr_list_head = nullptr;
+  }
+}
+
+swoc::Rv<struct ifaddrs *>
+InterfaceNameToEndpoint::find_matching_interface()
+{
+  swoc::Rv<struct ifaddrs *> zret{nullptr};
+  if (_ifaddr_list_head == nullptr) {
+    if (getifaddrs(&_ifaddr_list_head) == -1) {
+      zret.error("getifaddrs failed: {}", swoc::bwf::Errno{});
+      return zret;
+    }
+  }
+  for (auto *ifa = _ifaddr_list_head; ifa != nullptr; ifa = ifa->ifa_next) {
+    std::string_view interface_name{ifa->ifa_name};
+    if (interface_name != _expected_interface) {
+      continue;
+    }
+    auto const family = ifa->ifa_addr->sa_family;
+    if (family != _expected_family) {
+      continue;
+    }
+    zret = ifa;
+    return zret;
+  }
+  return nullptr;
+}
+
+swoc::Rv<std::string>
+InterfaceNameToEndpoint::convert_ip_endpoint_to_string(swoc::IPEndpoint const &ip)
+{
+  swoc::Rv<std::string> zret;
+  char ip_buffer[INET6_ADDRSTRLEN];
+  char const *ntop_ret = nullptr;
+  auto const family = ip.family();
+  if (family == AF_INET) {
+    ntop_ret = inet_ntop(family, &ip.sa4.sin_addr, ip_buffer, sizeof(ip_buffer));
+  } else { // AF_INET6
+    ntop_ret = inet_ntop(family, &ip.sa6.sin6_addr, ip_buffer, sizeof(ip_buffer));
+  }
+  if (ntop_ret == nullptr) {
+    zret.error(
+        "inet_ntop failed to convert the interface address to a string: {}",
+        swoc::bwf::Errno{});
+  }
+  zret = std::string{ip_buffer, strlen(ip_buffer)};
+  return zret;
+}
```

Comment:
I talked to Alan. This whole function can be removed with the caller having:

```
std::string ip_string;
swoc::bwprint(ip_string, "{}", ip_endpoint);
```

Very nice.